mantidproject/mantid
# Option to save default layout workbench

**Description of work.**
I have implemented an option to save the current layout of the workbench widgets.
On startup workbench will load widgets in the state they were when last closed.
Layout can be reset to any user to using option under view.

**To test:**
1. Open Workbench.
1. Change widget dock layout.
1. ``View --> User Layouts --> Settings``
1. Save your layout.
1. Test that the layout you just saved is accessable in the settings lesection box.
1. Check that your saved layout is listed under ``View --> User Layouts``
1. ``View --> Restore default layout``
1. Test that the option to restore your layout works from the settings menu and from ``View --> User Layouts --> [your layout name]``.
1. Change widget dock layout.
1. Restart Workbench.
1. Check that the the Workbench layout restores the last layout state.
1. Check that your layout is still available.
1. Delete your layout from the settings menu
1. Check that the layout is removed from the settings selection box and ``View --> User Layouts``.

Fixes #25515.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
